aptos/token_bridge/sources/bridge_state.move

```diff
@@ -118,13 +148,127 @@ module token_bridge::bridge_state {
+        let (burn_cap, freeze_cap, mint_cap) = initialize<CoinType>(&coin_signer, utf8(name), utf8(symbol), decimals, monitor_supply);
+
+        // update the following two mappings in State
+        // 1. (native chain, native address) => wrapped address
+        // 2. wrapped address => (native chain, native address)
+        let token_address = bridge_structs::get_token_address(& _asset_meta);
+        let token_chain = bridge_structs::get_token_chain(& _asset_meta);
+        let native_info = OriginInfo {token_address: token_address, token_chain: token_chain};
+        set_native_info(token_address, &native_info);
+        set_wrapped_asset(&native_info, token_address);
+
+        // store coin capabilities
+        let _token_bridge_signer = token_bridge_signer();
+        let coin_caps = CoinCapabilities<CoinType> { mint_cap: mint_cap, freeze_cap: freeze_cap, burn_cap: burn_cap};
+        move_to(&_token_bridge_signer, coin_caps);
+
+        vaa::destroy(vaa);
+    }
+
+    // transfer a native or wraped token from sender to token_bridge
+    public entry fun transfer_tokens<CoinType>(coins: Coin<CoinType>, relayer_fee: u128): TransferResult acquires State {
+        assert!(coin::is_account_registered<CoinType>(@token_bridge), E_COIN_NOT_REGISTERED);
+
+        let token_address = hash_type_info<CoinType>();
+        let amount = coin::value<CoinType>(&coins);
+
+        // transfer tokens from sender to token_bridge
+        coin::deposit<CoinType>(@token_bridge, coins);
+
+        // return TransferResult encapsulating details of token transferred
+        let origin_chain;
+        let origin_address;
+
+        if (is_wrapped_asset(token_address)) {
+            let _native_info = native_info(token_address);
+            origin_chain = _native_info.token_chain;
+            origin_address = _native_info.token_address;
+        } else {
+             if (!is_registered_native_asset(token_address)) {
+                set_native_asset(token_address, type_of<CoinType>());
+             };
+            origin_chain = get_chain_id();
+            origin_address = token_address;
+        };
+
+        // TODO - normalize amount by using helpers from utils.move
+        let normalized_amount = u256::from_u64(amount);
+        // TODO - normalize relayer fee
+        let normalized_relayer_fee = u256::from_u128(relayer_fee);
+        let wormhole_fee = u256::from_u64(0);
+
+        let transfer_result: TransferResult = create_transfer_result(
+            origin_chain,
+            origin_address,
+            normalized_amount,
+            normalized_relayer_fee,
+            wormhole_fee
+        );
+        transfer_result
+    }
+
+    public entry fun transfer_tokens_with_signer<CoinType>(sender: &signer, amount: u64, relayer_fee: u128): TransferResult acquires State {
+        let coins = coin::withdraw<CoinType>(sender, amount);
+        transfer_tokens<CoinType>(coins, relayer_fee)
+    }
+
+    public fun log_transfer(
```

Comment:
this being public seems like a major security risk

Reply:
made this public(friend)